frontend/app/utils/fetch-utils.ts

```diff
@@ -40,3 +41,25 @@ export function getFetchFn(proxyUrl?: string, timeout?: number) {
   log.debug('No proxy configured; using global fetch');
   return global.fetch;
 }
+
+/**
+ * Wraps a fetch() function and adds instrumentation/metrics collection
+ *
+ * @param fetchFn An existing fetch() function used to make the actual request
+ * @param metricPrefix A string prefix used for naming metrics (ex. "http.client.example-api.example-endpoint.gets", "http.client.example-api.example-endpoint.posts").
+ * @param input The URL or request object specifying the target of the fetch request
+ * @param init An object containing additional options for the fetch request (ex. headers, method)
+ * @returns A Promise that resolves to the Response object from the fetch() call
+ * @throws The original error thrown by the underlying fetch() call
+ */
+export async function instrumentedFetch(fetchFn: FetchFunction, metricPrefix: string, input: string | URL, init?: FetchFunctionInit) {
```

Comment:
I don't think `fetchFn` is reqired since your function signature already have `fetchFn` arguments.

Reply:
`fetchFn` is needed to distinguish whether the consumer passed `global.fetch` or `undiciFetch(..)` for when a proxy is used.

Reply:
Gotcha